src/main/java/seedu/duke/commands/CheckoutCommand.java

```diff
@@ -28,10 +32,14 @@ public void setToCheckout(Person person) {
     }
 
     @Override
-    public CommandOutput execute(TrackingList trackingList) {
-        toCheckout = trackingList.findExactPerson(id);
-        toCheckout.setCheckedIn(false);
-        return new CommandOutput(String.format(CHECKOUT_MESSAGE, toCheckout), COMMAND);
+    public CommandOutput execute(TrackingList trackingList) throws PersonNotFoundException {
+        toCheckout = trackingList.findExactPerson(toCheckout.getId());
+        if (toCheckout == null) {
+            throw new PersonNotFoundException();
+        } else {
+            toCheckout.setCheckedIn(false);
```

Comment:
I don't think you need to else part. You can just setCheckIn(false) outside the if without the else statement to make the happy path prominent.